packages/components/src/custom-select-control-v2/custom-select.tsx

```diff
@@ -52,7 +53,10 @@ const CustomSelectButton = ( {
 	...restProps
 }: Omit<
 	WordPressComponentProps<
-		CustomSelectButtonProps & CustomSelectButtonSize & CustomSelectStore,
+		CustomSelectButtonInternalProps &
+			CustomSelectButtonProps &
+			CustomSelectButtonSize &
+			CustomSelectStore,
 		'button',
 		false
 	>,
```

Comment:
Somewhat tangential, but maybe this should've composed from `Ariakit.Select` props rather than `<button>` props. Then we wouldn't need to add `CustomSelectButtonInternalProps` definitions for whatever other Ariakit props we want to pass. (No strong opinion, but probably cleaner in the long run.)

```diff
diff --git a/packages/components/src/custom-select-control-v2/custom-select.tsx b/packages/components/src/custom-select-control-v2/custom-select.tsx
index 144ccda7af..0520e392b9 100644
--- a/packages/components/src/custom-select-control-v2/custom-select.tsx
+++ b/packages/components/src/custom-select-control-v2/custom-select.tsx
@@ -1,3 +1,9 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type * as Ariakit from '@ariakit/react';
+
 /**
  * WordPress dependencies
  */
@@ -12,13 +18,11 @@ import * as Styled from './styles';
 import type {
 	CustomSelectContext as CustomSelectContextType,
 	CustomSelectStore,
-	CustomSelectButtonInternalProps,
 	CustomSelectButtonProps,
 	CustomSelectButtonSize,
 	_CustomSelectInternalProps,
 	_CustomSelectProps,
 } from './types';
-import type { WordPressComponentProps } from '../context';
 import InputBase from '../input-control/input-base';
 import SelectControlChevronDown from '../select-control/chevron-down';
 
@@ -52,14 +56,10 @@ const CustomSelectButton = ( {
 	store,
 	...restProps
 }: Omit<
-	WordPressComponentProps<
-		CustomSelectButtonInternalProps &
-			CustomSelectButtonProps &
-			CustomSelectButtonSize &
-			CustomSelectStore,
-		'button',
-		false
-	>,
+	React.ComponentProps< typeof Ariakit.Select > &
+		CustomSelectButtonProps &
+		CustomSelectButtonSize &
+		CustomSelectStore,
 	'onChange'
 > ) => {
 	const { value: currentValue } = store.useState();

```